src/acl/ServerName.h

```diff
@@ -20,12 +19,15 @@ class ACLServerNameData : public ACLDomainData {
     bool match(const char *) override;
 };
 
-class ACLServerNameStrategy : public ACLStrategy<char const *>
+namespace Acl
 {
 
+/// an "ssl::server_name" or "ssl::server_name_regex" ACL
+class ServerNameCheck: public ParameterizedNode< ACLData<const char *> >
```

Comment:
>  What do you think about having a code level namespace matching the ssl:: config namespace? I think it would be a good idea so that we can have different FooCheck classes for the distinct protocol scopes.

Yeah, I was thinking about namespaces for ACLs as well. I suspect that we are using the Acl namespace wrong -- it should not be simply/automatically strapped on top of anything that has to do with ACL-derived code. Instead, it should only be used for code that provides ACL fundamentals to the rest of Squid code base -- an Acl namespace is for the ACL "library" or "service" (in general sense, not the literal "libacls.a" library and such), not the code _using_ that library or service.

If we switch to that approach, many of the ACL-derived classes (including many of those touched in in this PR and others!) should be moved from their Acl namespace into the global namespace (or, in some rare cases, into other namespaces). They should also be moved outside the `src/acl` directory! IMO, those moves should be done in dedicated PR (or PRs), moving all classes (or, for example, moving all protocol Foo-related classes).

For example, this Acl::ServerNameCheck class may become ServerNameCheck (because it checks names using some non-TLS sources of information) or TlsSeverNameCheck (because it starts with or focuses on TLS sources of information). Note that it should not become Security::ServerNameCheck because the same reasoning that applies to the Acl namespace applies to the Security namespace as well.

N.B. I do not think we should focus on matching squid.conf namespaces with C++ namespaces. The two (obviously related) concepts operate in rather different environments, facing very different users and constraints. When there is a good match, great. When there is not, it is not a big deal. In case of `ssl::`, there is not a good match because Squid code uses `ssl` namespace for narrow, legacy reasons.

I did not want to discuss any of that in this PR because the topic is complex, controversial, and orthogonal to this PR changes. If we try to solve the two problems at once in this PR, we will reduce the number of times the same lines of code will be (trivially) modified, but we will muddy the important changes that this PR is focusing on and will significantly delay progress. I am certain that those changes, if any, should be done in a separate PR or PRs, after we develop consensus regarding namespace use. For now, this PR simply converts ACLFooStrategized into Acl::FooCheck and that is a good step forward as far as names are concerned, regardless of future changes, even if some of those future changes move all of the affected classes elsewhere.

----

> Sine this PR is including these renames.

Just to avoid a misunderstanding: This PR renames ACLFooStrategy classes because the corresponding code lines must be modified for in-scope reasons and because the old names are clearly bad, on several levels. I can undo all the ACLFooStrategy renaming, but it will not have a significant effect on the diff noise (and may complicate future class moves away from the Acl namespace).

Reply:
I think we are in rough agreement on the direction of ACL cleanup and use of `namespace Acl`. Future cleanup PRs/discussions can clarify further.